src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,484 @@
+  }
+
+  sync_primitives::Lock lock_;
+  NiceMock<MockHmiInterfaces> hmi_interfaces_;
+  MockMessageHelper& mock_message_helper_;
+  MockAppPtr mock_app_;
+};
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_NotRegisteredApplication_APPLICATION_NOT_REGISTERED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  ApplicationSharedPtr mock_app_empty;
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app_empty));
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest, Run_NaviApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::REJECTED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  NaviAppSetup();
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_NotMediaApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::REJECTED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  NonAppSetup();
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_VoiceCommunicationApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::REJECTED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  VoiceCommunicationAppSetup();
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_IndicatorWaitingForResponse_IGNORED) {
```

Comment:
@tkireva Consider renaming to `Run_AddIndicatorWaitingForResponseFailed_IGNORED`.